init/systemd/agent_start_script.sh

```diff
@@ -1,20 +1,6 @@
 #!/bin/sh
 
-# The variable ETH_IFACE contains the name the systemd gave to the network interface. 
-# The systemd configures the name based on the QEMU parameters. 
-# The parts of the name enp0s2 mean:
-#  et - ethernet card. It means this is the ethernet interface.
-#  p - means that the interface is connected to a PCI bus.
-#  0 - the interface is connected to bus 0.
-#  s2 -the interface is connected to slot 2.
-
-# The variable ETH_IFACE value must match the name configured through QEMU parameters for the network device. 
-# The bus number and slot number are configured through QEMU device parameters, parameters
-# addr (for slot number), and bus (for bus number).
 ETH_IFACE=enp0s2
```

Comment:
Instead of hardcoding this to `enp0s2`, can we use some command - like `if config`, or `addr` to get the interface name here, as it has been setup previously.

Reply:
If we call `dhclient` without giving it a interface name, then `dhclient` will initialize all of the network interfaces. Then we can use `route | grep '^default' | grep -o '[^ ]*$'` to fetch the name of the default interface. We only have one network interface (defined by QEMU) so this one will be the default interface. 

So, `ETH_IFACE=$(route | grep '^default' | grep -o '[^ ]*$')` but we first have to initialize all of the network interfaces by calling `dhclient`.

Reply:
After `dhclient` I can probably check if there is only one interface. To make sure that there is only one interface.

Reply:
Done